internal/controller/capacity.go

```diff
@@ -63,7 +63,8 @@ func (r *InstasliceReconciler) findNodeAndDeviceForASlice(ctx context.Context, i
 				updatedInstaSliceObject.Spec.PodAllocationRequests = make(map[types.UID]inferencev1alpha1.AllocationRequest)
 			}
 
-			newStart := r.getStartIndexFromPreparedState(updatedInstaSliceObject, gpuuuid, profileName)
+			gpuAllocatedIndex := gpuAllocatedSlices(updatedInstaSliceObject, gpuuuid)
+			newStart := r.getStartIndexFromPreparedState(updatedInstaSliceObject, profileName, gpuAllocatedIndex)
```

Comment:
I vote for doing minimal changes to the code logic, `getStartIndexFromPreparedState` should already have gpuAllocatedIndex, can we return the updated array from the method?